vsock_proxy/src/starter.rs

```diff
@@ -45,17 +45,35 @@ pub fn check_allowlist(
             .as_vec()
             .ok_or("No allowlist field")?;
 
+        // Obtain the remote server's IP address.
+        let mut addrs = Proxy::parse_addr(&remote_host, only_4, only_6)
+            .map_err(|err| format!("Could not parse remote address: {}", err))?;
+        let remote_addr = *addrs.get(0).ok_or("No IP address found")?;
```

Comment:
Why do we check only the first one, shouldn't we check all possible ip-addresses for a given hostname ?

Reply:
This was the behavior before this change and in practice there aren't any reported issues with it, so I didn't see a need to change it.

Reply:
Are we doing this for every packet ? If yes than we might incur a performance penalty, which probably we don't want to pay every packet. 

So what I would do would be:
1. Keep the list of allowed ips cached.
2. If we don't find the IP in the list of allowed ip, refresh the list and check again.
3. Let's also log somewhere when allowlist returns false, so we are able to easily debug issues later on.

Reply:
This is not done for every packet - it's performed only once, when the vsock-proxy starts. If the allowlist matching fails, the vsock-proxy won't start at all.